core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -429,7 +429,7 @@ class MapUnit {
 
         if(!hasUnique("All healing effects doubled") && type.isLandUnit() && type.isMilitary())
         {
-            val gainDoubleHealPromotion = tile.neighbors.filter{it.naturalWonder == "Fountain of Youth"}.any()
+            val gainDoubleHealPromotion = tile.neighbors.any{it.containsUnique("Grants Rejuvenation (all healing effects doubled) to adjacent military land units for the rest of the game")}
```

Comment:
why not by name? Seems like only `Fountain of Youth` has `Grants Rejuvenation...`

Reply:
> TL;DR: Anywhere that a nat wonder is referenced by name outside of the map generation, that's a place that should be replaced by a 'unique'
i believe this is to decouple the natural wonder type and the uniques it provides. in a near future users might create their mods and change what uniques natural wonders (or custom terrains) can provide.

Reply:
oh I see
I am just afraid of that we can easily make some mistakes when we trying to identify uniqueness of tiles by their loooong description
may be we can introduce some key type for uniqueness beside description

Reply:
Maybe `Unique` could be a pair `<Key, Description>`

Reply:
can we in `Terrains.json` replace `Unique` by `UniqueKey`(int) for example and make another `UniqueDescription.json` which will be like <UniqueKey, UniqueDescription>?
in this case we can operate in code only by `UniqueKey` and parse description only when we should render it on screen

Reply:
We could also use a common interface for all kind of uniques? Buildings, Units, Terrains etc.

Reply:
If I understand you correctly we can have 'uniqueKey' in buildings, units, terrain etc. But it should be different set of uniqueness for each type(for buildings, for units etc)
Otherwise in case of mods we can get huge set of uniqueness for all types and it will be hard to match them

Reply:
Let's not shoot ourselves in the foot with complexity =\
I want to ensure that things aren't referred to by name because this A. ensures that the term is displayable to users and translatable and B. the unique can be used by other things, e.g. in mods
A separate json for unique keys to their description will just complicate things